sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -952,9 +925,35 @@ export class MessageReceiver extends LinkEntity {
           connectionHost: this._context.namespace.config.host,
           delayInSeconds: 15
         };
-        if (!this.wasCloseInitiated) {
-          await retry<void>(config);
-        }
+        await retry<void>(config);
+      } else if (connectionDidDisconnect && !this.wasCloseInitiated) {
```

Comment:
Why not early exit from `onDetached()` if `this.wasCloseInitiated` is true?

Reply:
Updated to exit early if `this.wasCloseInitiated` is true.